app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -96,8 +96,13 @@ def self.with_environment(environment, include_default = false)
               where(katello_capsule_lifecycle_environments: { lifecycle_environment_id: environment.id })
         end
 
+        def self.pulpcore_proxy_with_environment(environment)
+          unscoped.where(id: unscoped.select { |p| p.pulp_mirror? }.pluck(:id)).joins(:capsule_lifecycle_environments).
+              where(katello_capsule_lifecycle_environments: { lifecycle_environment_id: environment.id })
+        end
+
         def self.sync_needed?(environment)
-          Setting[:foreman_proxy_content_auto_sync] && unscoped.with_environment(environment).any?
+          Setting[:foreman_proxy_content_auto_sync] && (unscoped.with_environment(environment).any? || pulpcore_proxy_with_environment(environment).any?)
```

Comment:
i kinda think we should either:  a) make with_environment() check both pulpcore and pulp2, or b) rename with_environment() to something like  'pulp2_with_environment()' 

also it look like with_environment is used in a couple  other places:
```
app/lib/actions/katello/content_view/capsule_sync.rb:              smart_proxies = SmartProxy.with_environment(environment)
app/lib/actions/katello/repository/capsule_sync.rb:              smart_proxies = ::SmartProxy.with_environment(repo.environment)
```